detect_secrets/plugins/telegram_token.py

```diff
@@ -0,0 +1,31 @@
+"""
+This plugin searches for Telegram bot tokens
+"""
+import re
+
+import requests
+
+from ..constants import VerifiedResult
+from detect_secrets.plugins.base import RegexBasedDetector
+
+
+class TelegramBotTokenDetector(RegexBasedDetector):
+    """Scans for Telegram bot tokens."""
+    secret_type = 'Telegram Bot Token'
+
+    denylist = [
+        # refs https://core.telegram.org/bots/api#authorizing-your-bot
+        re.compile(r'\d{8,10}:[0-9A-Za-z_-]{35}'),
+    ]
+
+    def verify(self, secret: str) -> VerifiedResult:  # pragma: no cover
```

Comment:
Looking at other `RegexBasedDetector, it doesn't look like you need a `verify` method like you would in a `BasePlugin` (see `github_token.py` for reference).

Reply:
[plugins.md#verified-secrets](https://github.com/Yelp/detect-secrets/blob/master/docs/plugins.md#verified-secrets)
> provides the ability to decrease false positives by making an API call to a server to check whether a PotentialSecret is indeed real. This can be extremely useful for specific secrets (e.g. RegexBasedDetector subclasses) since we are able to determine which server to contact.


`verify` method is optional but really good in decreasing `False-positives`. Others `RegexBasedDetector` plugins also has this method implemented where there is a way to verify whether the token is still active. ([stripe.py](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/plugins/stripe.py), [slack.py](https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/plugins/slack.py))

In my test in local:
- without verify: will report if any string matches the plugin regex ("is_verified": false) 
- with verify: will only report if the token is still active ("is_verified": true)